### PR TITLE
55467 - Create Password Settings Page 

### PR DESCRIPTION
**Pull request by RobertNugent04**
Create style and layout for Password Settings Page & add general input validation:

Screenshot:
passSett

### CODE REVIEW

**Comment by gursimar03** [comment]
Please Rename the title of the PR to "55467 - Create Password Settings Page"

File: upflux-client/src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
         <Routes>
-          <Route 
-            path="/" 
+          {/* Home route */}
+          <Route
+            path="/"
             element={
               <>
-                <Navbar />
+                <Navbar home={true} />
```
Comment (gursimar03) [inline upflux-client/src/App.tsx]: `home` doesnt provide any meaning to tell us what it is please change it to completing more meaningful. Thanks

File: upflux-client/src/features/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
 import logo from "../../assets/logos/logo-no-name.png";
 import './navbar.css';
 
-export const Navbar = () => {
+// Accept a prop `home` to conditionally render the navigation items
+interface NavbarProps {
+  home: boolean;
+}
+
+export const Navbar: React.FC<NavbarProps> = ({ home }) => {
```
Comment (gursimar03) [inline upflux-client/src/features/Navbar/Navbar.tsx]: ` export const Navbar: React.FC<NavbarProps> = ({ home }) =>` Change to 

```
export const Navbar: React.FC<NavbarProps> = (props : NavbarProps) => {
 const { home } = props;

}
```

also change home name to something else as mentioned above

File: upflux-client/src/features/PasswordSettings/PasswordSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  
+  const [errors, setErrors] = useState({
+    newPassword: '',
+    confirmPassword: '',
+  });
+
+  // Validation function for the New Password
```
Comment (gursimar03) [inline upflux-client/src/features/PasswordSettings/PasswordSettings.tsx:28]: Create a file in PasswordSettings Folder 

PasswordSettingsConsts.ts

in that 

```
export const MIN_PASSWORD_LENGTH = 12;
export const HAS_NUMBER_REGEX = /\d/;
... etc
export const PASSWORD_SET_NOT_ENOUGH_LENGTH = "Password must be at least 12 ....";
etc
```

do same for all "magic numbers" and "strings".

File: upflux-client/src/features/PasswordSettings/PasswordSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        onChange={(e) => setNewPassword(e.target.value)}
+        error={errors.newPassword}
+      />
+      <TextInput
+        label={
+          <>
+            Confirm new password <span className="label-required">*</span>
+          </>
+        }
+        placeholder="Please confirm your new password"
```
Comment (gursimar03) [inline upflux-client/src/features/PasswordSettings/PasswordSettings.tsx]: move this to const file

File: upflux-client/src/App.tsx [hunk 5]
```diff
@@ -28,10 +30,25 @@ export const App = () => {
                 </section>
                 <Footer />
               </>
-            } 
+            }
           />
 
+          {/* Login route */}
           <Route path="/login" element={<LoginComponent />} />
+
```
Comment (gursimar03) [inline upflux-client/src/App.tsx:18]: for the next PR or if you like. you can do here itself. Create File called LoginRoute.tsx and just in your return put the "Routing".. and use those components and fill up your app.tsx and we go forward we would be having a lot of routes and we do not want to clog up app.tsx file.

My suggestion is you should do it now.

**Review by gursimar03 — CHANGES_REQUESTED** [review]

**Comment by gursimar03** [comment]
Have you tested your UI on tablet view as well? to check its responsiveness?

File: upflux-client/src/routes/PasswordSettingsRoute.tsx
Comment (gursimar03) [inline upflux-client/src/routes/PasswordSettingsRoute.tsx:1]: Please move these files into there own folders.

**Review by gursimar03 — CHANGES_REQUESTED** [review]

**Review by Erlinguss — DISMISSED** [review]

**Review by gursimar03 — APPROVED** [review]

**Review by PatrickOrjieh — APPROVED** [review]

**Pull request merged**